classes/class-wpcom-liveblog-amp.php

```diff
@@ -503,4 +510,17 @@ public static function get_template( $name, $variables = array() ) {
 		$template = new WPCOM_Liveblog_AMP_Template();
 		return $template->render( $name, $variables );
 	}
+
+	/**
+	 * Is this an AMP polling request.
+	 *
+	 * @return boolean AMP polling request.
+	 */
+	public static function is_amp_polling() {
+		$amp_latest_update_time = filter_input( INPUT_GET, 'amp_latest_update_time', FILTER_SANITIZE_STRING );
```

Comment:
While the `filter` extension (that provides the `filter_input()` function) is enabled by default, it can also be disabled. See https://secure.php.net/manual/en/filter.installation.php

IIRC, WordPress core doesn't rely on this for this exact reason.

If this extension is to be a requirement, then it should be added to the `composer.json` `require` section, and also add a note to the READMEs.

Reply:
The `'amp_latest_update_time'` feels like a magic string - could it be added as a class constant instead?

Reply:
Hey @GaryJones 

Thanks for feedback. As I'm not using the value the ideal way would be:

```
	public static function is_amp_polling() {
		return isset( $_GET[ AMP_UPDATE_QUERY_VAR ] ); // input var okay.
	}
```
But that will result in `WARNING | Processing form data without nonce verification.`

I could add `filter` to `composer.json` but I guess trying to avoid using `filter` may make more sense if its not used anywhere else. The query var is added by the AMP plugin, unfortunately it doesn't appear possible to access it using `get_query_var()`

Reply:
If the nonce really isn't needed here (false positive), then the following should disable the warning:

```php
public static function is_amp_polling() {
	// phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
	return isset( $_GET[ AMP_UPDATE_QUERY_VAR ] );
}
```



> I guess trying to avoid using filter may make more sense if its not used anywhere else.

There are two existing uses of `filter_input()`, both in the root `liveblog.php` file. The predicament would obviously also apply to them, but since this PR wouldn't make the situation worse, I can't argue about it - it should have its own PR to discuss whether it should be a recognised dependency or not :-)